src/server/lib/Server.cc

```diff
@@ -100,14 +108,24 @@ void Server::activeRunLoop()
     m_runningNotifier.wait(lock, [this] { return !m_running.load(); });
 
     running = m_running.load();
+
+    std::cout << "Woken up, running = " << running << std::endl;
   }
 
-  m_systemProcessors.clear();
```

Comment:
This was not deleting the `SystemProcessor`s anymore as they are also stored as pointers in the `TriageMessageConsumer`. It means that the `SystemProcessor`s were actually continuing to process (and generate) messages until the very end.